### PR TITLE
Pie and Bar Chart Fix

### PR DESCRIPTION
**Pull request by AlexScheitlin**
Fixed pie and bar chart labels for too small pies and bars.

Depending on a threshold (manually set) the charts decide whether to display the label within or outside the pie/bar.

### CODE REVIEW

File: src/components/evaluation/charts/BarChart.js [hunk 1]
```diff
@@ -49,6 +49,25 @@ const calculatePercentage = (questionType, count, totalResponses) => {
   return ''
 }
 
+// break point for too small bars
+// if the percentual responses of a bar are smaller than the given
+// value, the label (A, B, ...)  is not displayed on top of the bar
+const smallBarBreak = 0.05
```
Comment (rschlaefli) [inline src/components/evaluation/charts/BarChart.js]: Should be defined as a constant in the constants file

File: src/components/evaluation/charts/BarChart.js [hunk 2]
```diff
@@ -49,6 +49,25 @@ const calculatePercentage = (questionType, count, totalResponses) => {
...
+
+const getPercentageIn = (questionType, count, totalResponses) => {
```
Comment (rschlaefli) [inline src/components/evaluation/charts/BarChart.js]: needs some comments

File: src/components/evaluation/charts/BarChart.js [hunk 3]
```diff
@@ -49,6 +49,25 @@ const calculatePercentage = (questionType, count, totalResponses) => {
...
+  if (count / totalResponses > smallBarBreak) {
+    return calculatePercentage(questionType, count, totalResponses)
+  }
+  return ''
+}
+
+const getPercentageOut = (questionType, count, totalResponses) => {
```
Comment (rschlaefli) [inline src/components/evaluation/charts/BarChart.js]: needs some comments

File: src/components/evaluation/charts/PieChart.js [hunk 4]
```diff
@@ -72,14 +72,42 @@ const PieChart = ({ isSolutionShown, data }) => (
 PieChart.propTypes = propTypes
 PieChart.defaultProps = defaultProps
 
+const indexToLetter = index =>
```
Comment (rschlaefli) [inline src/components/evaluation/charts/PieChart.js]: please extract these functions into a separate file (lib/charts.js). also the functions in the bar chart.. they could be reusable for something else, so we should extract them.

File: src/components/evaluation/charts/BarChart.js (hunk 1 above)
Comment (AlexScheitlin) [reply src/components/evaluation/charts/BarChart.js]: Done. I also extracted the variable for too small pies to the constant file. The new names are SMALL_BAR_THRESHOLD and SMALL_PIE_THRESHOLD.

File: src/components/evaluation/charts/BarChart.js (hunk 2 above)
Comment (AlexScheitlin) [reply src/components/evaluation/charts/BarChart.js]: Done

File: src/components/evaluation/charts/BarChart.js (hunk 3 above)
Comment (AlexScheitlin) [reply src/components/evaluation/charts/BarChart.js]: Done

File: src/components/evaluation/charts/PieChart.js (hunk 4 above)
Comment (AlexScheitlin) [reply src/components/evaluation/charts/PieChart.js]: Done :)

**Pull request merged**